src/plugins/Plugin.ts

```diff
@@ -15,28 +16,34 @@ export type PluginContext = {
     getSession: GetSession;
 };
 
-export interface Plugin {
+export abstract class Plugin {
+    protected enabled: boolean = true;
+
+    constructor(
+        public readonly pluginName: string,
+        private readonly pluginId = `${RUM_AWS_PREFIX}.${pluginName}`
```

Comment:
issue: A third party may also implement a plugin.

Reply:
Hmmmm maybe can make the prefix optional in the abstract?

Would that make sense? or make it so that it can be changed in the inherited class?

Reply:
but also, the pluginId can be passed as the 2nd param to the constructor (then this default value wont be used :-) )

Reply:
If we use an abstract class we would be enforcing implementation details, rather than simply defining behavior. This is ok for the built-in plugins, which we can change, but I don't think we should assume anything about how plugins will be implemented outside of this package.

For example, I may want to dynamically create my plugin's ID.

Reply:
and you still can


```ts

class BananaPlugin extends Plugin {
    constructor() {
        super('banana', someIdCreationMethod());
        // ...
    }

   // ...
}
```

Which will create a `BananaPlugin` with a name of `banana` and an id defined by them in the `someIdCreationMethod` method (or however they want to create it

Reply:
Unless I am misunderstanding you.

Reply:
Backwards compatibility is another issue. Existing third-party plugins could break.

Adding an abstract class on top of the interface will also give you more flexibility to make changes to the abstract class in the future, while if you replace the interface you will be limited in what you can do because of issues backwards compatibility.

Let's keep this as an interface.

Reply:
kk can do 

Will make an abstract class just for the internal plugins for now :-)